### PR TITLE
Add PR locking configuration at the digger.yml level

### PR DESCRIPTION
**Pull request by evanstachowiak**
Following the conversation here https://diggertalk.slack.com/archives/C01DVJRK6CX/p1718200181031169.

Adding the ability to enable/disable PR locking in GitHub with a digger.yml config setting.

Addresses this issue: https://github.com/diggerhq/digger/issues/1577

### CODE REVIEW

File: backend/controllers/github.go [hunk 1]
```diff
@@ -486,21 +486,23 @@ func handlePullRequestEvent(gh utils.GithubClientProvider, payload *github.PullR
...
-			Reporter:         comment_updater.NoopReporter{},
-			ProjectName:      project.Name,
-			ProjectNamespace: repoFullName,
-			PrNumber:         prNumber,
-		}
-		err = PerformLockingActionFromCommand(prLock, *diggerCommand)
-		if err != nil {
-			utils.InitCommentReporter(ghService, prNumber, fmt.Sprintf(":x: Failed perform lock action on project: %v %v", project.Name, err))
-			return fmt.Errorf("failed to perform lock action on project: %v, %v", project.Name, err)
+	if dg_configuration.PrLocks{
```
Comment (motatoes) [inline backend/controllers/github.go]: you need to use the config object here:

```suggestion
	if config.PrLocks{
```

Comment (motatoes) [inline backend/controllers/github.go]: you need to do the same if statement in the `handleIssueComment` webhook handler in the same file

**Review by motatoes — COMMENTED** [review]
Thanks for this. Overall looks good and I'm happy about the naming. 

We also need to override locking to be noopLock if this flag is set for backendless mode here: https://github.com/diggerhq/digger/blob/1d611b44e0af3efbb841c22fdd305a086f2474f7/cli/pkg/github/github.go#L214

**Comment by evanstachowiak** [comment]
I added support for `Enable` on the struct level, I couldn't see a better way to do it for the cli. In this case I could probably also remove these `if` statements around the `for` loops, but it would also save some processing to leave them in.

**Comment by motatoes** [comment]
Hey @evanstachowiak this looks great, thanks! I would rather not have a flag on the PrLock interface. We have a NoopLock interface implemented here: https://github.com/diggerhq/digger/blob/e43db334eaefb55c592bca7cc98d57fd524b74ca/libs/locking/locking.go#L36 which will ignore any lock operations

so if you had something like 

```
if config.Prlocks {
    log.Printf("info: using NoopLock)
    lock = NoopLock{}
}
```

[here](https://github.com/diggerhq/digger/blob/1d611b44e0af3efbb841c22fdd305a086f2474f7/cli/pkg/github/github.go#L214) it will do the trick - since that path is outside of the orchestrator path.

**Comment by motatoes** [comment]
@evanstachowiak actually thinking about it again, your changes will work well, the enable flag works. Future refactoring can be done but this is good enough for now!

**Review by motatoes — APPROVED** [review]

**Pull request merged**